docs/dev-client-networking.md

```diff
@@ -3,259 +3,647 @@ id: dev-client-networking
+entities it is the `owner` of.
+
+
+#### ClientLeave
+
+These tell a client that another client has disconnected. The next time the
+`networkReceiveSystem` runs, the receiving client will `remove` entities that
+the disconnected client was the `creator` of.
+
+
+#### CreatorChange
+
+These tell a client that the `creator` of an entity has been reassigned.
+Typically, this means that an entity has been `pinned` (or `unpinned`), and
+reticulum has assigned (or unassigned) itself as the entity&rsquo;s `creator`.
+
+
+### Ownership
+
+Most of the complexity in the `networkSendSystem` and the
+`networkReceiveSystem` stem from this property of the network. Here are some
+examples where this complexity reveals itself:
+
+- The `lastOwnerTime` is used to ensure that ownership transfer is handled
+identically by all clients, even when messages arrive out of order.
+- The `deletedNids` collection ensures that out-of-order `CreateMessage`s do
+not cause `deleted` entities to be accidentally recreated.
+- The `storedUpdates` allows a client to save `UpdateMessage` s it has received
+but has no way to process, as can happen when it receives an `UpdateMessage`
+from the `owner` of an entity before it receives a `CreateMessage` from its
+`creator`.
+- The `takeSoftOwnership` function allows clients to take ownership of unowned
+entities in such a way that only clients with the most recent information about
+that entity will be eligible as the new owner.
+
+For the most part, users of the networking systems do not need to understand
+these concepts. These are handled internally by the systems themselves. However,
+users do need to understand that ownership is not transactional or guaranteed.
+That is, ownership is not &ldquo;requested and then transferred&rdquo;, and
+just because one client claims ownership of an entity does not mean that other
+clients will respect that claim.
+
+Users can inspect the state the `Networked` or `Owned` components as needed in
+cases when their ownership claims matter. They may find themselves writing
+coroutines that looks like this:
+
+```typescript
+takeSoftOwnership(world, eid);
+yield sleep( 3000 ); // Wait a few seconds to see if we "win" ownership
+if (!hasComponent(world, Owned, eid)) return;
+```
 
-# Persisting Networked Entity State
+If this becomes a common and error-prone pattern, then we may introduce helper
+functions or additional semantics to cover these cases.
 
-By default, `network instantiated` entities are removed when its `creator` (a client) disconnects. In order to persist these entities (and its descendants) the entity must be `pinned`. Only `network instantiated` entities can be `pinned`.
 
-To `pin` an `network instantiated` entity, a client calls `createEntityState` . This will save the current state of the entity (and its descendants) to the database. We say that the entity (and its descendants) are `persistent`.
+### Async initialization

```

Comment:
There are two sections with this header.

Reply:
Updated, thanks